File: backend/models/src/response/departure.rs
use serde::Serialize;

use super::{station::ResStation, transfar::ResTransfar};

#[derive(Serialize, Debug, Clone, PartialEq)]
pub struct ResDeparture {
    pub ride_type: String,
    pub aka_type: Option<String>,
    pub type_foreground: String,
    pub type_background: String,
    pub type_pronounce: String,
    pub to: ResStation,
    pub career_type: String,
    pub depart_at: String,
    pub transfars: Vec<Vec<ResTransfar>>
}

File: frontend/composeApp/src/commonMain/kotlin/net/yourein/transfarnavi/models/Departure.kt
package net.yourein.transfarnavi.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Departure(
    @SerialName("ride_type") val rideType: String,
    @SerialName("aka_type") val akaType: String?,
    @SerialName("type_foreground") val typeForeground: String,
    @SerialName("type_background") val typeBackground: String,
    @SerialName("type_pronounce") val typePronounce: String,
    val to: Station,
    @SerialName("career_type") val careerType: String,
    @SerialName("depart_at") val departAt: String,
    @SerialName("transfars") val transfers: List<List<Transfer>>,
)
